packages/opencensus-web-instrumentation-zone/src/interaction-tracker.ts

```diff
@@ -319,15 +319,14 @@ export class InteractionTracker {
 
   private maybeUpdateInteractionName(previousLocationPathname: string) {
     const rootSpan = tracing.tracer.currentRootSpan;
-    // If for this interaction, the developer did not give any
-    // explicit attibute (`data-ocweb-id`) the current interaction
-    // name will start with a '<' that stands to the tag name. If that is
-    // the case, change the name to `Navigation <pathname>` as this is a more
-    // understadable name for the interaction.
-    // Also, we check if the location pathname did change.
+    // If for this interaction, the developer did not give any explicit
+    // attibute (`data-ocweb-id`) and the generated name is a CSS selector,
+    // that means the name might change to `Navigation <pathname>` as this is a
+    // more understadable name for the interaction in case the location
+    // pathname actually changed.
     if (
       rootSpan &&
-      rootSpan.name.startsWith('<') &&
+      document.querySelector(rootSpan.name) &&
```

Comment:
What's the role of this check? What would happen if it were just removed?

Reply:
As the `data-ocweb-id` attribute has the highest priority to name the interaction, and the `Navigation` is in second priority, we only want to rename the interaction as `Navigation` when the interaction was named using the CSS selector (third priority). Otherwise, if this is removed, the interaction will be named as `Navigation` even if the developer used the attribute `data-ocweb-id`.

Reply:
Hmm, OK. What if we just gave all interaction spans some temporary name (maybe just the empty string), and then do the logic for what to name the span at the end?

Reply:
mmm yeah, in that case it won't work as expected. What we can do is to check `rootSpan.name` is not empty, then we will know we are not overwriting the value in `data-ocweb-id`.